packages/context/src/notification/NotificationProvider.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, createContext, useEffect, useContext } from 'react';
+import { HttpTransportType, HubConnection, HubConnectionBuilder, LogLevel } from '@microsoft/signalr/dist/esm/index.js';
+import { useApiServices } from '../api/index.js';
+
+export type NotificationConnectionInfo = {
+    url: string;
+    accessToken: string;
+};
+
+export type Subscription<T> = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subscribe: (topic: string, callback: (...args: T[]) => any) => void;
```

Comment:
I think we need something more descriptive than `topic`.  E.g. if the intent is to subscribe to document changes for a specific object instance, then explicitly asking the caller to provide `objectId` and `instanceId` would make for an easier to use api.  Otherwise they have to know how to construct these topic names (btw what are examples of current topic names?).

Reply:
For document changes, we construct it as `${objectId}/${instanceId}`